### PR TITLE
{bio}[GCC/9.3.0] RAPSearch2 v2.24

### PR DESCRIPTION
**Pull request by manifestoso**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/42cc85df9dd54f07f69cbfe9e2feccdb for a full test report.

File: easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24-GCC-9.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+RAPSearch stands for Reduced Alphabet based Protein similarity Search
+"""
+
+toolchain = {'name': 'GCC', 'version': '9.3.0'}
+
+source_urls = ['https://sourceforge.net/projects/rapsearch2/files']
+sources = ['%(name)s.24_64bits.tar.gz']
+checksums = ['85db4573f4c768b6c3c73bb44ff2611ba48dc6c8d188feb40f44bf7c55de36ce']
+
+install_cmd = './install'
```
Comment (Micket) [inline easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24-GCC-9.3.0.eb:27]: Actually,  I don't think this is called at all here at all (perhaps it's not used when you specify files to copy)? I doesn't build anything, and you are just copying the prebuilt binaries. This actually depends on boost, so it wouldn't build without it.
I think you want `build_cmd` here (despite the name of this script "install", it actually just builds)

**Review by Micket — CHANGES_REQUESTED** [review]

**Comment by robqiao** [comment]
> 
> 
> Actually, I don't think this is called at all here at all (perhaps it's not used when you specify files to copy)? I doesn't build anything, and you are just copying the prebuilt binaries. This actually depends on boost, so it wouldn't build without it.
> I think you want `build_cmd` here (despite the name of this script "install", it actually just builds)

Good catch, `MakeCp` seems a better block for this. Thx @Micket,

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/4ac8365efae9c25dbfdcacb1278d1ee8 for a full test report.

File: easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+name = 'RAPSearch2'
+version = '2.24'
+
+homepage = 'https://omics.informatics.indiana.edu/mg/RAPSearch2'
+description = """
+RAPSearch stands for Reduced Alphabet based Protein similarity Search
+"""
+
+toolchain = {'name': 'system', 'version': 'system'}
```
Comment (verdurin) [inline easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24.eb:21]: @robqiao this is normally just `SYSTEM`:
```suggestion
toolchain = SYSTEM
```

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24.eb (hunk 2 above)
Comment (Micket) [reply easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24.eb:21]: Wait, why was this moved to system level at all? We are definitely compiling stuff here, the previous use of GCC-9.3.0 was definitely the correct choice.

Comment (robqiao) [reply easybuild/easyconfigs/r/RAPSearch2/RAPSearch2-2.24.eb:21]: Hmm, distribution tarball includes a legacy `boost` which is only compatible with legacy gcc 4.8.5
I guess we can try to link a modern `boost` but I prefer to use the distributed code as much as possible. See what you think @Micket

**Comment by robqiao** [comment]
> 
> 
> Wait, why was this moved to system level at all? We are definitely compiling stuff here, the previous use of GCC-9.3.0 was definitely the correct choice.


Hmm, distribution tarball includes a legacy `boost` which is only compatible with legacy gcc 4.8.5
I guess we can try to link a modern `boost` but I prefer to use the distributed code as much as possible. See what you think @Micket

**Comment by robqiao** [comment]
> 
> 
> There is no guarantee that the OS will have such an ancient compiler

That's true but as OS up to RHEL 7.9, `gcc` 4.8.x is available, maybe a note prompt in the header for the time being and wait until next software update? @Micket

**Comment by robqiao** [comment]
@verdurin @Micket @smoors can we merge this PR.

**Comment by verdurin** [comment]
@robqiao Please add some way of addressing the comment about compiler compatibility, so we can make a decision.

**Comment by robqiao** [comment]
> 
> 
> @robqiao Please add some way of addressing the comment about compiler compatibility, so we can make a decision.

@verdurin 
Done. The compiler issue was caused by `GCC-4.9.x` to `GCC-5.2.x` hard upgrade, i.e. `GCC-4.9.x` was not c++11 fully complied. Now switched to c++11 string compatible ABI. Case closed.

**Comment by robqiao** [comment]
@Micket @verdurin check this PR?

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/00143bad6e393b244d780c6bcd821ced for a full test report.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Review by verdurin — APPROVED** [review]

**Comment by verdurin** [comment]
@Micket I'm happy with this one now.

**Comment by manifestoso** [comment]
Finish this PR @smoors?

**Comment by verdurin** [comment]
Going in, thanks @manifestoso!

**Pull request merged**